Akka Persistence JDBC 5.0.0 release

octonato opened the pull request.

octonato commented:
release is stuck in sonatype, hold on on merging

johanandren commented on blog/_posts/2021-01-21-akka-persistence-jdbc-5.0.0-released.md:
```diff
@@ -0,0 +1,51 @@
+
+This release includes a new journal schema with tags persisted on a separated table.
+
+The serialized events are not wrapped with Akka’s `PersistentRepr` as before but directly saved as a blob payload into the column using Akka's serialization mechanism.
```
Perhaps worth noting the benefits? Easier inspection of data as it will only depend on user choice of serialization?

octonato replied:
I don't think we have much gain on easy inspection of data. This is true for Cassandra, because you can serialize a json payload and cql is capable to deserialize it. However, relational databases don't have it as far as I can tell. At the end, we do have opaque blob in the column. 

In Postgres it's possible to have a json column, but we don't support it.

johanandren replied:
If there is no benefit, why did we change it?

octonato replied:
There are benefits, but not for inspection. 

We can mention that `PersistentRepr` is an internal API, that although stable, could eventually change in the future. 

Also, now that I think more about it, there is another advantage. Although we can't inspect the payload, we can see which manifest was used to serialize it. Sort of inspection.

johanandren replied:
I like the announcements entries to be a little salespitchy "this new feature that you'll want because it gives you X", but do as you want with this.

johanandren commented on blog/_posts/2021-01-21-akka-persistence-jdbc-5.0.0-released.md:
```diff
@@ -0,0 +1,51 @@
+
+The serialized events are not wrapped with Akka’s `PersistentRepr` as before but directly saved as a blob payload into the column using Akka's serialization mechanism.
+
+ In order to migrate to the new schema, a migration tool capable of reading the serialized representation of `PersistentRepr` is required. That tool doesn’t exist yet, therefore, the **new schema can only be used with new applications**.
```
Is there an issue to link to for the community who wants to contribute that tool?

johanandren commented on blog/_posts/2021-01-21-akka-persistence-jdbc-5.0.0-released.md:
```diff
@@ -0,0 +1,51 @@
+
+ In order to migrate to the new schema, a migration tool capable of reading the serialized representation of `PersistentRepr` is required. That tool doesn’t exist yet, therefore, the **new schema can only be used with new applications**.
+
+ **If you have an existing database you need to update the configuration or migrate your data to the new schema.** See [Migration](https://doc.akka.io/docs/akka-persistence-jdbc/5.0/migration.html)**.
```
But the section above said migration is not possible?

octonato replied:
Migration is possible if you build a tool for it. Maybe I should be more explicit.

octonato commented:
@johanandren, I added the link to the issue and removed the confusing sentence. I think it's clear enough that migration is only possible if there is a migration tool, so no need to repeat it.

johanandren reviewed (approved):
I still find it a little bit odd but the linked doc page is clear enough so I guess that's fine.

octonato commented:
> I still find it a little bit odd but the linked doc page is clear enough so I guess that's fine.

What do you mean by odd? That we briefly talk about migration in the announcement?

johanandren commented:
This part "If you have an _existing database_ you need to update the configuration or migrate your data to the new schema" and then in the docs and above, oh btw, if you have an existing db you cant migrate because there is no tool.

octonato commented:
@johanandren, please check latest commit a246e2f8a4060430c7a9325178a1dc10a20e8e9b. 

I re-worded the sentence in an attempt to make it more straightforward.
